internal/jujucommands/jujucommands.go

```diff
@@ -13,21 +13,46 @@ import (
 	"github.com/mitchellh/go-linereader"
 )
 
-type outputLine struct {
+var (
+	cmdPrefix = "juju"
+)
+
+// OutputLine represents a line of output from a juju command.
+type OutputLine struct {
 	Line string
 	Err  error
 }
 
-var (
-	cmdPrefix = "juju"
-)
+// CommandRunner is a struct that runs juju commands and JUJU_DATA directory.
+type CommandRunner struct {
+	prefix      string
+	jujuDataDir string
+}
+
+// NewCommandRunner creates a new CommandRunner with the specified command prefix.
+//
+// dataDir is the JUJU_DATA directory where juju commands will store their data.
+// It must be an ABSOLUTE path.
+func NewCommandRunner(dataDir string) *CommandRunner {
+	return &CommandRunner{
+		prefix:      cmdPrefix,
+		jujuDataDir: dataDir,
+	}
+}
+
+// JujuDataDir returns the JUJU_DATA directory used by the CommandRunner.
+//
+// This is exposed so that commands that may need to know the JUJU_DATA directory can access it.
+func (b *CommandRunner) JujuDataDir() string {
+	return b.jujuDataDir
+}
 
 // runJujuCmd runs a juju command with the given command string and JUJU_DATA directory.
 // It returns a channel that will receive output lines from the command's stdout and stderr.
 // The command is run in a separate goroutine, and the context can be used to cancel the command.
-func runJujuCmd(ctx context.Context, args []string, jujuDataDir string) (<-chan outputLine, error) {
-	cmd := exec.CommandContext(ctx, cmdPrefix, args...)
-	cmd.Env = append(cmd.Env, "JUJU_DATA="+jujuDataDir)
+func (b *CommandRunner) RunJujuCmd(ctx context.Context, args []string) (<-chan OutputLine, error) {
+	cmd := exec.CommandContext(ctx, b.prefix, args...)
```

Comment:
Looks like the tests might need an update since this is using the struct's value rather than the global var.

Reply:
The structure was using the global var (which is why it worked) but I've removed it now as you requested. I updated signature of NewCmdRunner to take a prefix.